Test/DurableTask.AzureStorage.Tests/TestTablePartitionManager.cs

```diff
@@ -0,0 +1,723 @@
+﻿//  ----------------------------------------------------------------------------------
+//  Copyright Microsoft Corporation
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  ----------------------------------------------------------------------------------
+
+#nullable enable
+namespace DurableTask.AzureStorage.Tests
+{
+    using DurableTask.Core;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    [TestClass]
+    public class TestTablePartitionManager
+    {
+        readonly string connection = TestHelpers.GetTestStorageAccountConnectionString();
+
+        // Start with one worker and four partitions.
+        // Test the worker could claim all the partitions in 5 seconds.
+        [TestMethod]
+        public async Task TestOneWorkerWithFourPartition()
+        {
+            string testName = nameof(TestOneWorkerWithFourPartition);
+            var settings = new AzureStorageOrchestrationServiceSettings
+            {
+                TaskHubName = testName,
+                StorageConnectionString = this.connection,
+                WorkerId = "0",
+                AppName = testName,
+                UseTablePartitionManagement = true,
+            };
+            var service = new AzureStorageOrchestrationService(settings);
+            await service.StartAsync();
+
+
+            WaitForCondition(
+                timeout: TimeSpan.FromSeconds(5),
+                condition: () =>
+                {
+                    var partitions = service.ListTableLeases();
+                    Assert.AreEqual(4, partitions.Count());
+                    return partitions.All(p => p.CurrentOwner == "0");
+                });
+
+            await service.StopAsync();
+            await service.DeleteAsync();
+        }
+
+        //Starts with two workers and four partitions.
+        //Test that one worker can acquire two partitions. 
+        //Since two worker can not start at the same time, and one will start earlier than the another one.
+        //There would be a steal process, and test that the lease tranfer will take no longer than 30 sec.
```

Comment:
nit: let's add a space between `//` and the first character
also correct a few minor typos
```suggestion
        // Starts with two workers and four partitions.
        // Test that one worker can acquire two partitions,
        // since two workers can't start at the same time, and one will start earlier than the other one.
        // There should be a steal process, and test that the lease transfer will take no longer than 30 sec.
```